### ISSUE TITLE
Fix S8733 FP: Query-splitting behavior can be set globally

### ISSUE THREAD

**Issue by dougwaldron** [SonarSource/sonar-dotnet#9843]
### Description

Rather than configure query splitting for each individual query, you can [enable split queries globally](https://learn.microsoft.com/en-us/ef/core/querying/single-split-queries#enabling-split-queries-globally).

When this is set, SonarQube shouldn't trigger violations at the query level.

### Reproducer

Configure split queries globally:

```csharp
builder.Services.AddDbContext<AppDbContext>(db => db
    .UseSqlServer(connectionString, opts => opts.UseQuerySplittingBehavior(QuerySplittingBehavior.SplitQuery))
    .ConfigureWarnings(warnings => warnings.Throw(RelationalEventId.MultipleCollectionIncludeWarning))
);
```

Include a query in your app that would otherwise trigger this issue:

```csharp
var courses = await ctx.Courses
    .Include(c => c.Enrollments)
    .Include(c => c.Announcements)
    .ToListAsync();
```

### Product and Version

Both SonarQube Cloud and SonarAnalyzer.CSharp 10.29.0.143774
